app/crud/microserviceTeamScoreCard.py

```diff
@@ -0,0 +1,52 @@
+import sqlalchemy
+from sqlalchemy.orm import Session
+from ..models import Microservice
+from ..schemas import MicroserviceTeamScorecardBase,MicroserviceTeamScorecardCreate,MicroserviceTeamScorecardUpdate, team, scoreCard
+from .base import CRUDBase
+from typing import List
+from sqlalchemy.sql import func
+from starlette.exceptions import HTTPException
+from . import CRUDTeam, CRUDMicroserviceScoreCard, CRUDScoreCard
+
+
+
+class CRUDMicroserviceTeamScorecard(CRUDBase[MicroserviceTeamScorecardBase, MicroserviceTeamScorecardCreate, MicroserviceTeamScorecardUpdate ]):
+    def __init__(self, db_session: Session):
+        super(CRUDMicroserviceTeamScorecard, self).__init__(Microservice, db_session)
+        self.teamService= CRUDTeam(db_session)
+        self.scoreCardService= CRUDMicroserviceScoreCard(db_session) 
+        self.scoreCard = CRUDScoreCard(db_session) 
+
+    # Get all with scorecard and team
+    def getByServiceIdWithTeamAndSCOREDetails(self, service_id: int)-> MicroserviceTeamScorecardBase:
+    
+        microservice = self.get(service_id)
+    
+        if not microservice:
+            raise Exception(f"Microservice with id {service_id} not found")
+        
+        # Fetch the associated team
+        teamobject = self.teamService.get(microservice.teamId)  # Adjusted to match your model's attribute
+
+        # Fetch the scorecards associated with the microservice
+        scorecardIds = self.scoreCardService.getByServiceId(microservice.id)
+        scorecards = []
+        
+        for sc_id in scorecardIds:
```

Comment:
I believe you can write a custom function in the CRUDScorecard class, the function can write a custom sql filter query such that given a list of IDs it filters table records with id equal to one of the values in the provided list.
This way we query the table only once instead of repeating the query as many times as the length of the scorecardIds.

Please search this further and improve your code or prove this your current implementaion is more efficient : )

Reply:
I can see that there's already a function called `getByScoreCradIds`  in the CRUDScoreCard class why don't you use it ? : )